drivers/char/hw_random/bcm2835_rng_rust.rs

```diff
@@ -0,0 +1,31 @@
+// SPDX-License-Identifier: GPL-2.0
+
+//! Broadcom BCM2835 Random Number Generator support.
+
+#![no_std]
+#![feature(allocator_api, global_asm)]
+
+use alloc::boxed::Box;
+use core::pin::Pin;
+use kernel::prelude::*;
+use kernel::{cstr, platdev};
+
+module! {
+    type: RngModule,
+    name: b"bcm2835_rng_rust",
+    author: b"Rust for Linux Contributors",
+    description: b"BCM2835 Random Number Generator (RNG) driver",
+    license: b"GPL v2",
+}
+
+struct RngModule {
```

Comment:
For drivers that just register support for a platform device in their init function, we should provide a macro similar to C's `module_platform_driver`. I did that for `miscdev` in https://github.com/Rust-for-Linux/linux/blob/rust/rust/module.rs#L806.

This could perhaps be done in a different PR but I think it's important for us to provide this so that we reduce boilerplate code in Rust drivers, especially because their C counterparts already have such macros.

Reply:
Very good point !

Reply:
Can we defer this to the commit that adds `probe()`? We don't have a Trait object yet to pass to the macro.

Reply:
Yes, absolutely.